test/eth/TransactionManager.spec.js

```diff
@@ -208,4 +209,54 @@ describe('lifecycle hooks', () => {
     expect(biteHandlers.pending).toBeCalled();
     expect(biteHandlers.mined).toBeCalled();
   });
+
+  test('clear Tx when state is confirmed/finalized and older than 5 minutes', async () => {
+    const openId = uniqueId(open).toString();
+
+    const openHandlers = makeHandlers('open');
+    txMgr.listen(open, openHandlers);
+
+    // Add 10 minutes to the Tx timestamp
+    const myTx = txMgr._tracker.get(openId);
+    const minedDate = new Date(myTx._timeStampMined);
+    myTx._timeStampMined = minedDate.setMinutes(minedDate.getMinutes() + 10);
+
+    expect(txMgr._tracker._transactions).toHaveProperty(openId);
+
+    // after calling confirm, Tx state will become 'finalized' and be deleted from list.
+    await Promise.all([txMgr.confirm(open), mineBlocks(service)]);
+
+    expect(txMgr._tracker._transactions[openId]).toBeUndefined();
```

Comment:
This line appears to be causing the `Web3ProviderEngine does not support synchronous requests` error, that is causing the build to fail, although it is inconsistent when I run the test locally.

Reply:
Maybe try checking `Object.keys(txMgr._tracker._transactions).includes(openId)` instead--it could be that `expect` is aware that the value is a promise and then trying to resolve it (if it wasn't removed)